julia-mode.el

```diff
@@ -313,33 +301,21 @@
 (defconst julia-block-end-keywords
   (list "end" "else" "elseif" "catch" "finally"))
 
-(defun julia-stringify-triple-quote ()
-  "Put `syntax-table' property on triple-quoted string delimiters.
-
-Based on `python-syntax-stringify'."
-  (let* ((string-start-pos (- (point) 3))
-         (string-end-pos (point))
-         (ppss (prog2
-                   (backward-char 3)
-                   (syntax-ppss)
-                 (forward-char 3)))
-         (in-comment (nth 4 ppss))
-         (in-string (nth 8 ppss)))
-    (unless in-comment
-      (if in-string
-          ;; We're in a string, so this must be the closing triple-quote.
-          ;; Put | on the last " character.
-          (put-text-property (1- string-end-pos) string-end-pos
-                             'syntax-table (string-to-syntax "|"))
-        ;; We're not in a string, so this is the opening triple-quote.
-        ;; Put | on the first " character.
-        (put-text-property string-start-pos (1+ string-start-pos)
-                           'syntax-table (string-to-syntax "|"))))))
-
 (defconst julia-syntax-propertize-function
   (syntax-propertize-rules
-   ("\"\"\""
-    (0 (ignore (julia-stringify-triple-quote))))
+   ;; triple-quoted strings are a single string rather than 3
+   ((rx (group ?\") ?\" (group ?\"))
+    (1 (let ((ppss (save-excursion (syntax-ppss (match-beginning 0)))))
+         (unless (or (nth 3 ppss) (nth 4 ppss))
+           (string-to-syntax "|"))))
```

Comment:
Is `string-to-syntax` needed? I thought that `syntax-propertize-rules` did that automatically for strings.

Also, can you please add some comments to each branch? Eg here we apply the syntax if we are inside a comment or a string.

Why do we need the second `group`, is it used for anything?

(I am not trying to nitpick here, just trying to make the code easier to maintain for people less familiar with Emacs Lisp and conventions.)

Reply:
`string-to-syntax` is needed unless the `SYNTAX` part of `HIGHLIGHT1` is a string literal.

I'm pushing a commit with some additional comments.

The second group is used by the `(2 (...` expression to properly end triple-quoted-strings.